src/wp-includes/script-loader.php

```diff
@@ -3671,3 +3671,33 @@ function wp_enqueue_classic_theme_styles() {
 		wp_enqueue_style( 'classic-theme-styles' );
 	}
 }
+
+/**
+ * Loads classic theme styles on classic themes in the editor.
+ *
+ * This is needed for backwards compatibility for button blocks specifically.
+ *
+ * @since 6.1.0
+ *
+ * @param array $editor_settings The array of editor settings.
+ * @return array A filtered array of editor settings.
+ */
+function wp_add_editor_classic_theme_styles( $editor_settings ) {
+	if ( wp_is_block_theme() ) {
+		return $editor_settings;
+	}
+	$suffix = wp_scripts_get_suffix();
+	$classic_theme_styles = "/wp-includes/css/dist/block-library/classic$suffix.css";
```

Comment:
I think it's the leading slash. This might work fine for "smart" functions such as `wp_register_style`, but `file_get_contents` will take this literally -- for the file system root 😱

Reply:
This seems to do the trick:

```diff
diff --git a/src/wp-includes/script-loader.php b/src/wp-includes/script-loader.php
index 8f9e420d3c..fd91718ebf 100644
--- a/src/wp-includes/script-loader.php
+++ b/src/wp-includes/script-loader.php
@@ -3687,13 +3687,13 @@ function wp_add_editor_classic_theme_styles( $editor_settings ) {
                return $editor_settings;
        }
        $suffix = wp_scripts_get_suffix();
-       $classic_theme_styles = "/wp-includes/css/dist/block-library/classic$suffix.css";
+       $classic_theme_styles = "/css/dist/block-library/classic$suffix.css";
 
        // This follows the pattern of get_block_editor_theme_styles,
        // but we can't use get_block_editor_theme_styles directly as it
        // only handles external files or theme files.
        $editor_settings['styles'][] = array(
-               'css'            => file_get_contents( $classic_theme_styles ),
+               'css'            => file_get_contents( __DIR__ . $classic_theme_styles ),
                'baseURL'        => get_theme_file_uri( $classic_theme_styles ),
                '__unstableType' => 'theme',
                'isGlobalStyles' => false,
```

I don't know about [`get_theme_file_uri`](https://developer.wordpress.org/reference/functions/get_theme_file_uri/). Can that ever work as it's supposed to for a file inside `wp-includes`?

Reply:
Maybe we don't need the `baseURL` field?

Reply:
Fixed the path in d1b37eb33e9fc3e746a8e09cad78426c8e1d62b3, and removed `baseURL` in 1bc847445b2decc4948ba6fb8ae149bc4bed0b97.

Reply:
Thank you